02-peer-protocol.md

```diff
@@ -191,11 +188,11 @@ This message contains a transaction input.
 2. data:
     * [`channel_id`:`channel_id`]
     * [`u64`:`serial_id`]
-    * [`u16`:`prevtx_len`]
+    * [`bigsize`:`prevtx_len`]
     * [`prevtx_len*byte`:`prevtx`]
     * [`u32`:`prevtx_vout`]
     * [`u32`:`sequence`]
-    * [`u16`:`script_sig_len`]
+    * [`bigsize`:`script_sig_len`]
     * [`script_sig_len*byte`:`script_sig`]
```

Comment:
Why not drop this entirely? This is only used for segwit nested inside p2sh, right? Can't we just restrict to native segwit and remove that field entirely?

Reply:
We totally could. Not sure on the net benefit of removing it? I couldn't think of a valid reason to exclude wrapped-segwit inputs from being spent here haha.

Reply:
It's one less case to handle! And we want to push for a native segwit world anyway since we require outputs to be using native segwit, so it's more consistent.

Also, in the eclair case I cannot easily support wrapped-segwit anyway, because I'll ask bitcoind to do a `fundrawtransaction` but I don't have a good way of obtaining this `script_sig` (unless I ask them to sign before the transaction is completely built, but it's really wasteful).

Reply:
I went ahead and removed it in https://github.com/niftynei/lightning-rfc/pull/3/commits/37491624085d967f9a36372d9112d1d3589607e5, let me know if you find it acceptable! I also updated the requirements to be native-segwit only, we're in 2022 it's time!

Reply:
@t-bast's got it out for the wrapped segwit wallets still out there jkjk.  sgtm!

For the record, I'm not totally into the `bigsize` in places that don't need it but that's more of a stylistic complaint than a real one, so happy to defer to your preference here.

Reply:
> For the record, I'm not totally into the bigsize in places that don't need it but that's more of a stylistic complaint than a real one, so happy to defer to your preference here.

I don't feel very strongly about this either, let's poll the other implementers during the spec meeting to see what people like!